Update python.md with tracestate propagation and span links behavior (#20736)
DataDog/documentation: content/en/tracing/trace_collection/trace_context_propagation/python.md

If multiple environment variables are set, `DD_TRACE_PROPAGATION_STYLE_INJECT` and `DD_TRACE_PROPAGATION_STYLE_EXTRACT`
override any value provided in `DD_TRACE_PROPAGATION_STYLE`.

> When multiple extraction styles are enabled, the extraction attempt is done on the order those styles are configured,
> using the first successful extracted value. If later valid trace contexts are found, they are terminated and appended
> as span links. Moreover, if the tracecontext style is enabled, W3C Tracestate is propagated if W3C Traceparent matches
> the extracted context.

Reviewer: ```suggestion
When multiple extraction styles are enabled, the extraction attempt is done in the order those styles are configured
using the first successful extracted value. If valid trace contexts are found later, they are terminated and appended
as span links. Additionally, if the `tracecontext` style is enabled, W3C Tracestate is propagated if W3C Traceparent matches
the extracted context.
```
